public/app/features/explore/state/query.ts

```diff
@@ -696,6 +698,114 @@ export const runQueries = createAsyncThunk<void, RunQueriesOptions>(
   }
 );
 
+interface RunLoadMoreLogsQueriesOptions {
+  exploreId: string;
+  absoluteRange: AbsoluteTimeRange;
+}
+/**
+ * Supplementary action to run queries that request more results, and dispatches sub-actions based
+ * on which result viewers are active
+ */
+export const runLoadMoreLogsQueries = createAsyncThunk<void, RunLoadMoreLogsQueriesOptions>(
+  'explore/runQueries',
+  async ({ exploreId, absoluteRange }, { dispatch, getState }) => {
+    dispatch(cancelQueries(exploreId));
+
+    const correlations$ = getCorrelations(exploreId);
+    const { datasourceInstance, containerWidth, queryResponse, correlationEditorHelperData } =
+      getState().explore.panes[exploreId]!;
+
+    const isCorrelationEditorMode = getState().explore.correlationEditorDetails?.editorMode || false;
+    const isLeftPane = Object.keys(getState().explore.panes)[0] === exploreId;
+    const showCorrelationEditorLinks = isCorrelationEditorMode && isLeftPane;
+    const defaultCorrelationEditorDatasource = showCorrelationEditorLinks ? await getDataSourceSrv().get() : undefined;
+    const interpolateCorrelationHelperVars =
+      isCorrelationEditorMode && !isLeftPane && correlationEditorHelperData !== undefined;
+    let newQuerySource: Observable<ExplorePanelData>;
+
+    // Filter queries by those explicitly requested by refId
+    const logQueries = queryResponse.logsResult?.queries || [];
```

Comment:
Hmmm 🤔 this should be probably applied to all results, not only logs. Is it actually needed? We have some logic in urlSync to ensure data source is provided in the query.

Reply:
Sorry, the comment about:

> Is it actually needed? We have some logic in urlSync to ensure data source is provided in the query.

Referred to lines below:

```typescript
datasource: query.datasource || datasourceInstance?.getRef(),
```

I was wondering in what case `query.datasource` is missing.

Reply:
I'm not sure, it's just a copy of what is used in `runQuery`:

https://github.com/grafana/grafana/blob/master/public/app/features/explore/state/query.ts#L546